src/Fields/Fields.jl

```diff
@@ -51,4 +53,13 @@ end
 
 include("set!.jl")
 
+struct SumOfFields{N, F}
+    fields :: F
+    SumOfFields{N}(fields...) where N = new{N, typeof(fields)}(fields)
+end
+
+@inline getindex(s::SumOfFields{1}, i, j, k) = @inbounds s.fields[1][i, j, k]
```

Comment:
```suggestion
@propagate_inbounds getindex(s::SumOfFields{1}, i....) = getindex(s.fields[1], i...)
```

I believe that's what we want. Check out the docs for `Base.@propagate_inbounds`. (Note that we _really_ should use `propagate_inbounds` everywhere...)